File: backend/src/main/java/com/ihecride/controllers/EvaluationController.java
package com.ihecride.controllers;

import com.ihecride.dao.EvaluationDAO;
import com.ihecride.dao.UserDAO;
import com.ihecride.models.Evaluation;

import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;

@WebServlet({"/api/evaluations", "/api/evaluations/conducteur"})
public class EvaluationController extends HttpServlet {

    private final EvaluationDAO dao = new EvaluationDAO();
    private final UserDAO userDAO = new UserDAO();

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
        resp.setContentType("text/plain; charset=UTF-8");
        int evalueId = Integer.parseInt(req.getParameter("id"));
        List<Evaluation> list = dao.findByEvalue(evalueId);
        try (PrintWriter out = resp.getWriter()) {
            out.println("id|note|commentaire|evaluateurNom|date");
            for (Evaluation e : list) {
                out.println(e.getId() + "|" + e.getNote() + "|" +
                            safe(e.getCommentaire()) + "|" + safe(e.getEvaluateurNom()) + "|" + e.getDate());
            }
        }
    }

    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
        req.setCharacterEncoding("UTF-8");
        int userId = (int) req.getAttribute("userId");
        Evaluation e = new Evaluation();
        e.setEvaluateurId(userId);
        e.setEvalueId(Integer.parseInt(req.getParameter("evalueId")));
        e.setTrajetId(Integer.parseInt(req.getParameter("trajetId")));
        e.setNote(Integer.parseInt(req.getParameter("note")));
        e.setCommentaire(req.getParameter("commentaire"));
        int id = dao.insert(e);
        if (id > 0) userDAO.updateNoteMoyenne(e.getEvalueId());
        resp.getWriter().write("success=" + (id > 0));
    }

    private String safe(String s) { return s == null ? "" : s.replace("|", "/").replace("\n"," "); }
}

File: backend-php/controllers/EvaluationController.php
<?php
/**
 * Contrôleur évaluations : /api/evaluations, /api/evaluations/conducteur.
 * Équivalent PHP de com.ihecride.controllers.EvaluationController.
 */

require_once __DIR__ . '/../dao/EvaluationDAO.php';
require_once __DIR__ . '/../dao/UserDAO.php';
require_once __DIR__ . '/../models/Evaluation.php';

class EvaluationController {

    private EvaluationDAO $dao;
    private UserDAO $userDAO;

    public function __construct() {
        $this->dao     = new EvaluationDAO();
        $this->userDAO = new UserDAO();
    }

    public function byEvalue(array $ctx): void {
        $evalueId = (int) ($_GET['id'] ?? 0);
        json_send(['success' => true, 'data' => $this->dao->findByEvalue($evalueId)]);
    }

    public function create(array $ctx): void {
        $e = new Evaluation();
        $e->evaluateurId = $ctx['userId'];
        $e->evalueId     = (int) ($_POST['evalueId'] ?? 0);
        $e->trajetId     = (int) ($_POST['trajetId'] ?? 0);
        $e->note         = (int) ($_POST['note']     ?? 0);
        $e->commentaire  = $_POST['commentaire'] ?? '';
        $id = $this->dao->insert($e);
        if ($id > 0) $this->userDAO->updateNoteMoyenne($e->evalueId);
        json_send(['success' => $id > 0]);
    }
}
